src/imp/cgroup.c

```diff
@@ -168,5 +169,41 @@ struct cgroup_info *cgroup_info_create (void)
     return cgroup;
 }
 
+int cgroup_kill (struct cgroup_info *cgroup, int sig)
+{
+    int count = 0;
+    int rc = 0;
+    int saved_errno = 0;
+    char path [PATH_MAX+15]; /* cgroup->path + "/cgroup.procs" */
+    FILE *fp;
+    unsigned long child;
+    pid_t current_pid = getpid ();
+
+    (void) snprintf (path, sizeof (path), "%s/cgroup.procs", cgroup->path);
```

Comment:
Check for overflow?

Reply:
Here we know we've added enough space to the buffer so it can't overflow. I guess `sprintf` could be used with a note in that case?